lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -630,6 +631,50 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
     end
   end
 
+  @doc """
+  Provide randomness to the operation of the beacon chain.
+  """
+  @spec process_randao(BeaconState.t(), BeaconBlockBody.t()) ::
+          {:ok, BeaconState.t()} | {:error, binary}
+  def process_randao(
+        %BeaconState{} = state,
+        %BeaconBlockBody{randao_reveal: randao_reveal} = _body
+      ) do
+    epoch = Accessors.get_current_epoch(state)
+
+    # Verify RANDAO reveal
+    with {:ok, proposer_index} <- Accessors.get_beacon_proposer_index(state) do
+      proposer = Enum.at(state.validators, proposer_index)
+      domain = Accessors.get_domain(state, Constants.domain_randao(), nil)
+      signing_root = Misc.compute_signing_root(epoch, SszTypes.Epoch, domain)
+
+      if Bls.valid?(proposer.pubkey, signing_root, randao_reveal) do
+        randao_mix = Accessors.get_randao_mix(state, epoch)
+        hash = :crypto.hash(:sha256, randao_reveal)
+
+        # Mix in RANDAO reveal
+        mix =
+          Enum.zip(:binary.bin_to_list(randao_mix), :binary.bin_to_list(hash))
+          |> Enum.map_join(fn {left, rigth} -> Bitwise.bxor(left, rigth) end)
```

Comment:
[`:crypto.exor`](https://www.erlang.org/doc/man/crypto#exor-2) can be used for this:
```suggestion
        mix = :crypto.exor(randao_mix, hash)
```